From f738aea78bc986fdfba8e28562229be256f393b0 Mon Sep 17 00:00:00 2001
From: Bo Yang <bo.yang@amlogic.com>
Date: Mon, 15 Jun 2015 20:33:01 +0800
Subject: [PATCH 5655/5965] PD#107797: dts: n301: add support for i2c-a

Add support for i2c-a for new referrence board of t868.

In new refference board of t868(20150511V02), the tuner use i2c-a.
If you use the old board of t868, you should changed the adapter id
for tuner,
	tuner0_i2c_adap_id = <1>;
the <1> is i2c-a, and <2> is i2c-b, etc...

Change-Id: I9d4880dbb17af68663a5d1ff6d701c0c3957eb37
---
 arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd | 33 ++++++++++++++++++--
 1 file changed, 31 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd b/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd
index 7c20b9df0ea7..c3b99ba76d59 100644
--- a/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd
+++ b/arch/arm/boot/dts/amlogic/mesong9tv_n301.dtd
@@ -674,6 +674,26 @@
 
 /// ****************************************************************************
 ///	-	I2C
+//$$ MODULE="I2C"
+//$$ DEVICE="I2C_A"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2="reg"
+//$$ L3 PROP_STR = "pinctrl-names"
+//$$ L2 PROP_CHOICE "I2C_A_pin_match" = "pinctrl-0"
+	i2c@c1108500{ /*I2C-A*/
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-A";
+		status = "ok";
+		reg = <0xc1108500 0x20>;
+		device_id = <1>;
+		pinctrl-names="default";
+		pinctrl-0=<&a_i2c_master>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		use_pio = <0>;
+		master_i2c_speed = <300000>;
+	};
+
 //$$ MODULE="I2C"
 //$$ DEVICE="I2C_B"
 //$$ L2 PROP_STR = "status"
@@ -1531,6 +1551,16 @@
 			amlogic,pullupen = <1>;
 		};
 
+
+		//$$ MATCH "I2C_A_pin_match" = "&a_i2c_master"
+		//$$ L2 PROP_U32 2 = "amlogic,setmask"
+		//$$ L2 PROP_STR 2 = "amlogic,pins"
+		a_i2c_master:a_i2c {
+			amlogic,setmask = <6 0xc0000000>;
+			amlogic,clrmask = <7 0x06000000>;
+			amlogic,pins = "GPIOW_0","GPIOW_1";
+		};
+
 		//$$ MATCH "I2C_B_pin_match" = "&b_i2c_master"
 		//$$ L2 PROP_U32 2 = "amlogic,setmask"
 		//$$ L2 PROP_STR 2 = "amlogic,pins"
@@ -1538,7 +1568,6 @@
 			amlogic,setmask = <10 0xc000>;
 			amlogic,clrmask = <9 0xf00000c0>;
 			amlogic,pins = "GPIOY_12","GPIOY_13";
-
 		};
 
 		//$$ MATCH "remote_pin_0_match" = "&remote_pins"
@@ -1683,7 +1712,7 @@
 	//	fe0_mem = <0>;
 	//	reserve-memory = <0x2000000>;  //32m
 		tuner0 = "si2151_tuner";
-		tuner0_i2c_adap_id = <2>;
+		tuner0_i2c_adap_id = <1>;
 		tuner0_i2c_addr = <0x60>;
 		tuner0_reset_value = <0>;
 		tuner0_reset_gpio =  "GPIOW_1" ;  /*GPIOX_8   76*/
-- 
2.19.0

